helper.go

```diff
@@ -273,18 +273,19 @@ func arrayToKV(m map[string]RedisMessage, arr []RedisMessage, keys []string) map
 
 func clusterMGet(client Client, ctx context.Context, keys []string) (ret map[string]RedisMessage, err error) {
 	ret = make(map[string]RedisMessage, len(keys))
-	slots := make(map[uint16][]int, len(keys)/2)
-	for i, key := range keys {
-		s := slot(key)
-		slots[s] = append(slots[s], i)
+	slotCount := make(map[uint16]int, len(keys)/2)
+	for _, key := range keys {
+		slotCount[slot(key)]++
 	}
-	cmds := mgetcmdsp.Get(0, len(slots))
+	cmds := mgetcmdsp.Get(0, len(slotCount))
 	defer mgetcmdsp.Put(cmds)
-	groups := make([][]string, 0, len(slots))
-	for _, group := range slots {
-		gkeys := make([]string, 0, len(group))
-		for _, i := range group {
-			gkeys = append(gkeys, keys[i])
+	groups := make([][]string, 0, len(slotCount))
+	for s, count := range slotCount {
+		gkeys := make([]string, 0, count)
```

Comment:
hmm, ideally, we should avoid `gkeys := make([]string, 0, count)` too. We should only do allocations on `client.B().Mget().Key(...)`